pkg/asset/agent/image/oc.go

```diff
@@ -88,19 +89,29 @@ func (r *release) GetBaseIso(log logrus.FieldLogger, releaseImage string, pullSe
 	return path, err
 }
 
-func (r *release) getImageFromRelease(log logrus.FieldLogger, imageName, releaseImage, pullSecret string, haveMirror bool) (string, error) {
+func (r *release) getImageFromRelease(log logrus.FieldLogger, imageName, releaseImage, pullSecret string, mirrorConfig []mirror.RegistriesConfig) (string, error) {
 	// This requires the 'oc' command so make sure its available
 	_, err := exec.LookPath("oc")
 	if err != nil {
-		if haveMirror {
+		if len(mirrorConfig) > 0 {
 			log.Warning("Unable to validate mirror config because \"oc\" command is not available")
 		} else {
 			log.Debug("Skipping ISO extraction; \"oc\" command is not available")
 		}
 		return "", err
 	}
 
-	cmd := fmt.Sprintf(templateGetImage, imageName, true, releaseImage)
+	if len(mirroConfig) > 0 {
+		log.Debugf("Using mirror configuration")
+		icspFile, err := getIcspFileFromRegistriesConfig(log, mirrorConfig)
+		if err != nil {
+			return "", err
+		}
+		defer removeIcspFile(icspFile)
+		cmd := fmt.Sprintf(templateGetImageWithIcsp, imageName, true, icspFile, releaseImage)
+	} else {
+		cmd := fmt.Sprintf(templateGetImage, imageName, true, releaseImage)
+	}
```

Comment:
```suggestion
	cmd := fmt.Sprintf(templateGetImage, imageName, true, releaseImage)
	if len(mirrorConfig) > 0 {
		log.Debugf("Using mirror configuration")
		icspFile, err := getIcspFileFromRegistriesConfig(log, mirrorConfig)
		if err != nil {
			return "", err
		}
		defer removeIcspFile(icspFile)
		cmd = fmt.Sprintf(templateGetImageWithIcsp, imageName, true, icspFile, releaseImage)
	}
```

Reply:
mirroConfig -> mirrorConfig
cmd needs to initialized outside of 
`if len(mirrorConfig) > 0 {`

Reply:
thanks. I just edited that and testing again

Reply:
I had the go mod cache gone. So I had to download the whole thing before the build (so pushed early)